packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -69,11 +70,17 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
   let [isCollapsed, setIsCollapsed] = useState(maxRows != null);
   let state = useListState(props);
   let [tagState, setTagState] = useValueEffect({visibleTagCount: state.collection.size, showCollapseButton: false});
-  let keyboardDelegate = useMemo(() => (
-    isCollapsed
-      ? new TagKeyboardDelegate(new ListCollection([...state.collection].slice(0, tagState.visibleTagCount)), direction)
-      : new TagKeyboardDelegate(new ListCollection([...state.collection]), direction)
-  ), [direction, isCollapsed, state.collection, tagState.visibleTagCount]) as TagKeyboardDelegate<T>;
+  let keyboardDelegate = useMemo(() => {
```

Comment:
during resizing of the tag group container, this will be regenerated twice due to useValueEffect
Maybe we could independently memo both keyboard delegates?

Reply:
is it any different than it was before though?

Reply:
true, just noticed it now